docs/source/traits_user_manual/listening.rst

```diff
@@ -0,0 +1,786 @@
+.. _on-trait-change-notification:
+
+===========================================
+Trait Notification with **on_trait_change**
+===========================================
+
+**on_trait_change** is an older method for setting up change notifications. It
+has several design flaws and limitations. A newer mechanism **observe** is
+introduced for overcoming them. See :ref:`observe-notification` for details and
+for how to migrate.
+
+Change notifications can be set up with **on_trait_change** in several ways:
```

Comment:
This sentence and the paragraph above are changed. The rest are the same as the original documentation.